Update review doc (#1350)
avohq/docs: pages/data-design/branches/index.mdx

### Creating a branch

To create a new branch, first make sure you are on the main branch. Click the branch name in the left side navigation bar and then click the “Create Branch” button.

![Create branch](/images/data-design/quickstart/create-branch.png)

### Request branch review

> When you've suggested changes to your tracking, mark the branch as ready for review, and assign someone from your team to review the changes.

Reviewer: ```suggestion
When you've suggested changes to your tracking, mark the branch as ready for review, and assign an editor or admin to review the changes.
```
